cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerSecurityService.java

```diff
@@ -313,6 +330,45 @@ public String getMasterKey() {
         return securityConfigProvider.getMasterKey(stack.getCluster());
     }
 
+    @Override
+    public void rotateHostCertificates() throws CloudbreakException {
```

Comment:
could you refactor this into smaller parts or move the logic into a separate class?